### PR TITLE
Fix l1 regularization bug

### PR DESCRIPTION
**Pull request by chengduoZH**
For sparse parameter regularization, we should extract the corresponding parameters from the origin parameters, and add regularization on the extracted parameter but not the origin parameter.

### CODE REVIEW

File: python/paddle/fluid/regularizer.py [hunk 1]
```diff
@@ -237,6 +237,7 @@ def __call__(self, param, grad, block):
                         'Ids': idx},
                 outputs={'Out': decay},
                 attrs={'is_sparse': True})
+            param = decay
```
Comment (jacquesqiao) [inline python/paddle/fluid/regularizer.py:240]: After change, the 
```python
block.append_op(
             type='sign', inputs={"X": param}, outputs={"Out": decay})	             type='sign', inputs={"X": param}, outputs={"Out": decay})
```
in and out will be the same, is this for purpose?

Comment (chengduoZH) [reply python/paddle/fluid/regularizer.py:240]: Yes

**Comment by jacquesqiao** [comment]
Add a unit test?

**Review by jacquesqiao — APPROVED** [review]
please remember to add a unit test

**Pull request merged**